js/app/src/Blocks.svelte

```diff
@@ -234,6 +234,21 @@
 	let handled_dependencies: Array<number[]> = [];
 
 	let messages: (ToastMessage & { fn_index: number })[] = [];
+	const add_message = (
+		message: string,
+		fn_index: number,
+		type: ToastMessage["type"]
+	) => {
+		messages = [
+			{
+				message,
+				fn_index,
+				type,
+				id: ++_error_id
+			},
+			...messages
+		];
+	};
```

Comment:
Can we refactor this to be pure (not modify outside state and just return a value), I'd like to start doing this more in the future so we may as well start. It will allow us to pull more stuff outside of the svelte files + make them easier to unit tests.

Something like :
```js
const create_message = (
  message: string,
  fn_index: number,
  type: ToastMessage["type"]
) => ({
    message,
    fn_index,
    type,
    id: ++_error_id
});
```

Then we could use it like:

```js
messages = [ create_message(_message, fn_index, "error"), ...messages];
```

Or we can pass in the original `message` array as well and actually concatenate them in the fn:

```js
messages = add_message(messages, _message, fn_index, "error");
```

Reply:
I don't mind which.